### PR TITLE
Ignore downlevel browsers for NewRelic monitoring

### PR DESCRIPTION
**Pull request by philbogle**
Don't perform NewRelic JS monitoring for Internet Explorer version 9 or lower, for both dashboard and pegasus.  

We don't support old IE browsers and want to ignore them for purposes of alerting and reporting.   We exclude those browsers by wrapping the NewRelic scripts with a "downlevel revealed"    conditional comment.  The conditional comment causes old IE browsers to ignore the scripts, but is ignored by newer versions of IE and all other browsers.

The wrapping is done by monkey patching the NewRelic method that adds the NewRelic script tags; this ensures that the wrapping occurs both in auto-injection and for manually inserted instrumentation.  I'm not generally a fan of monkey patching but in this case it is the most reliable way to make sure the change applies in all possible edge cases.  Worst case, if the monkey patching stops working, we will start monitoring old IE versions again which is relatively benign.

NewRelic monitoring normally occurs only in production.  I have exercised this change as much as possible in development (with some additional tweaks to allow it apply in development) for both Pegasus and Dashboard but final verification can happen only in production.  ( A heads up to @davidsbailey.)  I verified that the conditional comment is included when NewRelic is enabled for both Pegasus and Dashboard.  I have also verified that IE9 does in fact ignore the content inside the conditional comment and that Chrome correctly evaluates it.

### CODE REVIEW

**Comment by philbogle** [comment]
@bcjordan FYI

**Comment by davidsbailey** [comment]
Sounds good overall in terms of other New Relic monitoring which I am aware of. Ideally we wouldn't turn off custom events and page actions for downlevel browsers, but I doubt it's worth the eng effort to preserve those.

File: dashboard/config/application.rb [hunk 1]
```diff
@@ -128,5 +128,10 @@ class Application < Rails::Application
 
     # turn off ActionMailer logging to avoid logging email addresses
     ActionMailer::Base.logger = nil
+
+    if Rails.env.production?
+      require 'newrelic_rpm'
```
Comment (wjordan) [inline dashboard/config/application.rb:133]: My hunch is that this line is a no-op, as the above `Bundler.require(:default, Rails.env)` (on line [20](https://github.com/code-dot-org/code-dot-org/pull/8354/files#diff-5fdb0cec2f0205f0e50d1df7631fb1e4R20)) will already automatically `require` all gems not marked `require: false` referenced by our `Gemfile` for the specified groups.

However, it's not doing any harm to explicitly include it here, so should be fine to leave as-is (you'd need to do another full round of testing if it's removed).

**Comment by wjordan** [comment]
LGTM

**Comment by philbogle** [comment]
Thanks for the review.

**Pull request merged**